src/lib/conditions.ts

```diff
@@ -0,0 +1,49 @@
+type TConditionCallback = () => boolean;
```

Comment:
```suggestion
type TConditionCallback = (() => boolean) | boolean;
```
This might bother you, but I do wish to use booleans or callbacks here for all of `C`, if you oppose it don't do it but I think it could be useful.